File: app/src/main/java/kz/orkenniet/home/presentation/HomeFragment.java
package kz.orkenniet.home.presentation;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;

import kz.orkenniet.R;

public class HomeFragment extends Fragment {

    public HomeFragment() {
        // Required empty public constructor
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_home, container, false);
        return view;
    }
}

File: app/src/main/java/kz/orkenniet/home/presentation/HomeFragment.kt
package kz.orkenniet.home.presentation

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import by.kirich1409.viewbindingdelegate.viewBinding
import kz.orkenniet.R
import kz.orkenniet.core.extentions.navigate
import kz.orkenniet.databinding.FragmentHomeBinding

class HomeFragment : Fragment(R.layout.fragment_home) {

    private val binding: FragmentHomeBinding by viewBinding(FragmentHomeBinding::bind)

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        setupNavigation()
    }

    private fun setupNavigation() = with(binding) {
        llAdventureGenre.setOnClickListener {
            navigateToSortedBook("приключения")
        }
        llDetectiveGenre.setOnClickListener {
            navigateToSortedBook("детектив")
        }
        llFairyTailsGenre.setOnClickListener {
            navigateToSortedBook("сказки")
        }
    }

    private fun navigateToSortedBook(genre: String) {
        navigate(
            HomeFragmentDirections.actionHomeFragmentToSortedBookFragment(genre)
        )
    }
}

